File: lib/data/summary.dart
class Summary{

  Map<String, dynamic> data;
  Summary(this.data);

  bool isRunning(){
    if(data==null || !data.containsKey("isRunning")) return false;
    return data["isRunning"];
  }
  int getDeviceNo(){
    if(data==null || !data.containsKey("deviceNo")) return 0;
    return data["deviceNo"];

  }
  String getRunningTime(){
    if(data==null || !data.containsKey("runningTime")) return '';
    int runningTime = data["runningTime"];
    runningTime = (runningTime/1000).round();
    int sec = runningTime%60;
    runningTime = (runningTime/60).round();
    int min = runningTime%60;
    int hour = (runningTime/60).round();
    String res = (hour.toString().padLeft(2, '0'))+':'+(min.toString().padLeft(2, '0'))+":"+(sec.toString().padLeft(2, '0'));
    return res;
  }
  List getDevices(){
    if(data==null || !data.containsKey("sensorSummary")) return new List();
    return data["sensorSummary"];
  }
}

File: android/src/main/java/org/md2k/motionsenselib/summary/Summary.java
package org.md2k.motionsenselib.summary;

import org.md2k.motionsenselib.device.Device;
import org.md2k.motionsenselib.device.MotionSenseManager;
import org.md2k.motionsenselib.device.SensorInfo;

import java.util.ArrayList;

public class Summary {
    private long runningTime;
    private boolean isRunning;
    private int deviceNo;
    private ArrayList<SensorSummary> sensorSummary;
    public static Summary create(){
        Summary summary = new Summary();
        summary.runningTime = MotionSenseManager.getRunningTime();
        summary.isRunning = summary.runningTime != -1;
        summary.deviceNo = MotionSenseManager.getDeviceNo();
        summary.sensorSummary = new ArrayList<>();
        ArrayList<Device> devices = MotionSenseManager.getDevices();
        for(int i=0;i<devices.size();i++){
            ArrayList<SensorInfo> sensors = devices.get(i).getSensorInfo();
            for(int j=0;j< sensors.size();j++){
                SensorSummary s = new SensorSummary();
                s.count=sensors.get(j).getCount();
                s.deviceId = devices.get(i).getDeviceSettings().getDeviceId();
                s.platformType = devices.get(i).getDeviceSettings().getPlatformType();
                s.platformId = devices.get(i).getDeviceSettings().getPlatformId();
                s.lastSampleTime = sensors.get(j).getLastSampleTime();
                s.lastSample = sensors.get(j).getLastSample();
                s.sensorType = sensors.get(j).getSensorType().name();
                s.sensorName = sensors.get(j).getTitle();
                if(sensors.get(j).getStartSampleTime()>0){
                    s.frequency = (s.count*1000.0)/(System.currentTimeMillis()-sensors.get(j).getStartSampleTime());
                }else s.frequency = 0;
                summary.sensorSummary.add(s);
            }
        }
        return summary;
    }
}
